### PR TITLE
[5.8] [stdlib] Update the stdlib to use Unicode 15 data

### PR DESCRIPTION
**Pull request by Azoy**
Cherry pick of https://github.com/apple/swift/pull/64731 to the 5.8 branch.

Resolves: https://github.com/apple/swift/issues/63754

### CODE REVIEW

**Comment by Azoy** [comment]
@swift-ci please test

File: validation-test/stdlib/StringGraphemeBreaking.swift [hunk 1]
```diff
@@ -102,7 +102,7 @@ extension _StringGuts {
   func _isForeign() -> Bool
 }
 
-if #available(SwiftStdlib 5.6, *) {
+if #available(SwiftStdlib 5.9, *) {
```
Comment (stephentyrone) [inline validation-test/stdlib/StringGraphemeBreaking.swift:105]: Is `SwiftStdlib 5.9` availability defined on the 5.8 branch?

**Comment by Azoy** [comment]
@swift-ci please test

File: validation-test/stdlib/StringGraphemeBreaking.swift (hunk 1 above)
Comment (Azoy) [reply validation-test/stdlib/StringGraphemeBreaking.swift:105]: It is now 😃

**Review by stephentyrone — APPROVED** [review]

**Pull request merged**